pkgs/development/python-modules/unsloth-zoo/default.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  setuptools,
+  setuptools-scm,
+  torch,
+  packaging,
+  transformers,
+  tyro,
+  datasets,
+  sentencepiece,
+  tqdm,
+  psutil,
+  accelerate,
+  trl,
+  peft,
+  huggingface-hub,
+  hf-transfer,
+  cut-cross-entropy,
+
+}:
+
+buildPythonPackage rec {
+  pname = "unsloth-zoo";
+  version = "2025-02-v2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    repo = "unsloth-zoo";
+    # owner = "unslothai";
+    # rev = "31fce522e291e95c3829dc8cc8ac29e03e6c0580";
+    # sha256 = "sha256-GGh5B1JF+Q0nRhpUqq7EkwN8X55dUtcZ1/9nmyYpmfQ=";
+    # For patching the requirement for deprecated protobuf<4
+    owner = "hoh";
```

Comment:
Depending on your fork for this would be unnecessary, you can just use `postPatch` after `src`:
```
postPatch = ''
  substituteInPlace pyproject.toml \
      --replace-fail "protobuf<4.0.0" "protobuf"
'';
```